packages/plugin-text/src/plugins/katex/index.tsx

```diff
@@ -0,0 +1,373 @@
+import {
+  NodeRendererProps,
+  NodeEditorProps,
+  TextPlugin,
+  NodeControlsProps
+} from '../..'
+import { debounce } from 'lodash'
+import * as React from 'react'
+import { Block, Editor, Inline, BlockJSON, InlineJSON } from 'slate'
+import { isHotkey } from 'is-hotkey'
+
+import { Math } from './math.component'
+import { OverlayContext } from '@edtr-io/core'
+import { Button, Checkbox, Overlay, Textarea } from '@edtr-io/ui'
+
+// @ts-ignore
+import MathQuill, { addStyles as addMathquillStyles } from 'react-mathquill'
+
+addMathquillStyles()
+
+export const katexBlockNode = '@splish-me/katex-block'
+export const katexInlineNode = '@splish-me/katex-inline'
+
+export const isKatex = (editor: Editor) => {
+  return (
+    editor.value.blocks.some(block =>
+      block ? block.type === katexBlockNode : false
+    ) ||
+    editor.value.inlines.some(inline =>
+      inline ? inline.type === katexInlineNode : false
+    )
+  )
+}
+export const insertKatex = (editor: Editor) => {
+  return editor.insertInline({
+    type: katexInlineNode,
+    data: {
+      formula: '',
+      inline: true
+    }
+  })
+}
+
+export interface KatexPluginOptions {
+  EditorComponent?: React.ComponentType<NodeEditorProps>
+  RenderComponent?: React.ComponentType<NodeRendererProps>
+  ControlsComponent?: React.ComponentType<NodeControlsProps>
+}
+
+const DefaultEditorComponent: React.FunctionComponent<
+  NodeEditorProps
+> = props => {
+  const { attributes, node, editor } = props
+
+  const overlayContext = React.useContext(OverlayContext)
+
+  const { data } = node as Block | Inline
+  const inline = data.get('inline')
+  const formula = data.get('formula')
+  const setFormula = React.useState(
+    node ? node.data.get('formula') : undefined
+  )[1]
+
+  const handleChange = debounce((formula: string) => {
+    editor.setNodeByKey(node.key, {
+      type: node.type,
+      data: {
+        formula,
+        inline: node.data.get('inline')
+      }
+    })
+  }, 50)
```

Comment:
This looks pretty low. Maybe you don't need a debounce at all?

Reply:
debounce is removed